### ISSUE TITLE
Decide on heuristics for which cpr graphs on account / climber profile page

### ISSUE THREAD

**Issue by brendanheywood** [theCrag/website#2613]
I've started the work integrating the new graph into the account page

* [x] we need to rip out all the old chart data and make sure we haven't missed any db queries that could be slowing the page down. 

* [x] what are the heuristics which decide which graphs we will show, and in what order? Order should be easy, but not 100% about the if.

Do we simply show an full graph as soon as someone has a single tick, even aspiration, of a given gear style?

Would order by based on the volume of ticks per type, or on your cpr score ie show what you are climbing the hardest first?

**Comment by scd** [comment]
In regards to the heuristic for which cpr curve to show I have the following `curvePriorities` parameter

```
{
            'sliceCount' => 29,
            'name' => 'onsightCpr',
            'priority' => 1,
            'ascentCount' => 1210
          },
          {
            'sliceCount' => 29,
            'name' => 'redpointCpr',
            'priority' => 2,
            'ascentCount' => 2654
          },
          {
            'sliceCount' => 29,
            'name' => 'pinkpointCpr',
            'priority' => 10,
            'ascentCount' => 2899
          },
          {
            'sliceCount' => 29,
            'name' => 'flashCpr',
            'priority' => 12,
            'ascentCount' => 1859
          },
          {
            'sliceCount' => 23,
            'name' => 'topropeCpr',
            'priority' => 14,
            'ascentCount' => 109
          },
          {
            'sliceCount' => 16,
            'name' => 'secondCpr',
            'priority' => 15,
            'ascentCount' => 36
          }
```
The heuristic is based on the number of time slices that an ascent occurs of that type with extra processing for onsight and redpoint which are put up the top if either of them occur in more that 3 slices.

So deciding which curves to show is now a matter of iterating down this list until you find a suitable stopping criteria.

**Comment by brendanheywood** [comment]
ok thinking out loud, what if we have someone who's learning to climb and happily climbs inside for a while, then top ropes outside for ages etc. They get a nice increasing performance curve and then after a couple years they progress and start leading and get new curve and their old curve just disappears! Maybe we should entertain the idea of there being X curves at any given point rather than in the whole graph?

**Comment by scd** [comment]
> Maybe we should entertain the idea of there being X curves at any given point rather than in the whole graph

Nice idea. We need to make sure that we are not going backward and forward, so which ever curves we choose should be sticky. Maybe it is actually fairly easy. If you know you need to plot an onsight curve at some point, you can choose another curve until you reach that point. At this stage it is only top rope and seconds where this matters.

**Comment by scd** [comment]
@brendanheywood I am getting a strange artifact where the vertical scrollbar is always on for the iframe no matter what height I set. Is this something you have heard of before. If you don't just know about this then I will look into it, but have got to run right now.

**Comment by brendanheywood** [comment]
Yeah I just saw that and had a play but couldn't see anything obviously wrong and the expected fix didn't work, so more digging needed

**Comment by scd** [comment]
Sometimes the simple things soak up a lot of time :)

**Comment by scd** [comment]
I just fixed the vertical scrollbar height issue. What a pain - two things going on and both had to be fixed.
Firstly the svg should be display: block as there are some strange artifacts with defaults when inline-block, I think with extra white spaces.

Secondly and probably related, we don't want all our js and css for these aggregate images.

The useful thing about investigating this was that it brought up a couple of other issues:
 - [ ] do we want google analytics and facebook tracking for embeds. Currently disabled.
 - [ ] do we what all the head tag meta data, images etc. I have turned off most of this meta data for embeds.
